### PR TITLE
Refactor model save to include labels, metadata and custom_objects

### PR DESCRIPTION
**Pull request by larme**
- add `labels` and `custom_objects` to frameworks' `save` function
- add test cases for frameworks' `save`
- minor cleanup

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/frameworks/keras.py [hunk 1]
```diff
@@ -188,8 +182,9 @@ def save(
     model: "keras.Model",
     *,
     save_format: t.Optional[str] = "tf",
-    custom_objects: t.Optional[t.Dict[str, t.Any]] = None,
     store_as_json_and_weights: t.Optional[bool] = False,
+    labels: t.Optional[t.Dict[str, str]] = None,
+    custom_objects: t.Optional[t.Dict[str, t.Any]] = None,
```
Comment (parano) [inline bentoml/_internal/frameworks/keras.py:187]: should we differentiate Keras custom_objects with our model store custom_objects?

Comment (parano) [inline bentoml/_internal/frameworks/keras.py:187]: should we differentiate Keras custom_objects with our model store custom_objects?

File: tests/integration/frameworks/test_easyocr_impl.py [hunk 2]
```diff
@@ -43,8 +43,23 @@ def image_array():
 
 
 @pytest.fixture(scope="module")
-def save_proc() -> t.Callable[[t.Dict[str, t.Any], t.Dict[str, t.Any]], "Model"]:
-    def _(lang_list, recog_network, detect_model, metadata) -> "Model":
+def save_proc() -> t.Callable[
```
Comment (parano) [inline tests/integration/frameworks/test_easyocr_impl.py:46]: make this a function call instead of fixture

Comment (parano) [inline tests/integration/frameworks/test_easyocr_impl.py:46]: make this a function call instead of fixture

Comment (parano) [reply tests/integration/frameworks/test_easyocr_impl.py:46]:
```
def save_model(
         lang_list,
         recog_network,
         detect_model,
         metadata,
         labels=None,
         custom_objects=None,
     ) -> "Model":
```

Comment (parano) [reply tests/integration/frameworks/test_easyocr_impl.py:46]:
```
def save_model(
         lang_list,
         recog_network,
         detect_model,
         metadata,
         labels=None,
         custom_objects=None,
     ) -> "Model":
```

File: tests/integration/frameworks/test_picklable_model_impl.py [hunk 3]
```diff
@@ -25,12 +25,16 @@ class MockBatchOptions:
 
 def save_procedure(
```
Comment (parano) [inline tests/integration/frameworks/test_picklable_model_impl.py:26]: rename to just `save_model`

File: bentoml/_internal/frameworks/keras.py (hunk 1 above)
Comment (larme) [reply bentoml/_internal/frameworks/keras.py:187]: In my test we can mix them together. So saving keras custom_objects with other custom_objects won't affect how model is loaded.

We can also add a `keras_custom_objects` argument, but do you think it's a little bit cumbersome? @parano

**Comment by parano** [comment]
The python3.6 e2e test failed due to the missing docker image, not related to this PR
